rfcs/0127-issues-warnings.md

```diff
@@ -0,0 +1,156 @@
+---
+feature: issues-warnings
+start-date: 2022-06-11
+author: piegames
+co-authors: (find a buddy later to help out with the RFC)
+shepherd-team: (names, to be nominated and accepted by RFC steering committee)
+shepherd-leader: (name to be appointed by RFC steering committee)
+related-issues: https://github.com/NixOS/nixpkgs/pull/177272
+---
+
+# RFC: Nixpkgs issues and warnings
+
+## Summary
+[summary]: #summary
+
+Introduce an issue system into Nixpkgs, similar to broken and insecure, but with a custom per-package message. This will then be used to warn users about packages that are in need of maintenance. Packages that have an open issue for a long time should eventually be removed.
+
+## Motivation
+[motivation]: #motivation
+
+Nixpkgs has the problem that it is often treated as "append-only", i.e. packages only get added but not removed. There are a lot of packages that are broken for a long time, have end-of-life dependencies with known security vulnerabilities or that are otherwise unmaintained.
+
+Let's take the end of life of Python 2 as an example. (This applies to other ecosystems as well, and will come up again and again in the future.) It has sparked a few bulk package removal actions by dedicated persons, but those are pretty work intensive and prone to burn out maintainers. A goal of this RFC is to provide a way to notify all users of a package about the outstanding issues. This will hopefully draw more attention to abandoned packages, and spread the work load. It can also help soften the removal of packages by providing a period for users to migrate away at their own pace.
```

Comment:
> A goal of this RFC is to provide a way to notify all users of a package about the outstanding issues

And maintainers too I hope! I maintain some packages that I rarely use (like, yearly or so), so I would miss the user notification and would only get the removal notification.

Reply:
Ideally, we'd already have functioning pinging infrastructure in place so that this RFC would only apply to unmaintained packages. I know that at the moment we have some room for improvement on regarding maintainer pings, but that issue is out of scope for this RFC.

> notify all users of a package

Since the warnings happen at evaluation time, "users" refers to all people having the package installed or somehow in scope. So if you are a maintainer, it suffices to keep your packages installed.

Reply:
> So if you are a maintainer, it suffices to keep your packages installed.

I maintain ~80 packages, and I don't use all of them all the time from the unstable branch, so it would take a while for me to notice